cli_tools/gce_ovf_import/multi_image_importer/request_builder.go

```diff
@@ -0,0 +1,75 @@
+//  Copyright 2021 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package multiimageimporter
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/image/importer"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/path"
+	ovfdomain "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_import/domain"
+)
+
+type requestBuilder struct {
+	workflowDir   string
+	sourceFactory importer.SourceFactory
+}
+
+// buildRequests constructs a list of ImageImportRequests based on the user's
+// invocation parameters and the files referenced in the OVF descriptor.
+func (r *requestBuilder) buildRequests(params *ovfdomain.OVFImportParams, fileURIs []string) (requests []importer.ImageImportRequest, err error) {
+	timeout := params.Deadline.Sub(time.Now())
+	if timeout <= 0 {
+		return nil, errors.New("Timeout exceeded")
+	}
```

Comment:
Moved to the executor. (The original idea was that it was a validation, but I agree that the executor is a better location, since it's actually running the code)